contributors/design-proposals/volume-provisioning.md

```diff
@@ -412,9 +419,13 @@ type StorageClass struct {
   // Parameters for dynamic volume provisioner.
   Parameters map[string]string `json:"parameters,omitempty"`
 }
-
 ```
 
+StorageClass can have annotation
+`volume.beta.kubernetes.io/storage-class-selectors` that contains free-form
+english text with description of labels that are supported by the provisioner
+in pvc.spec.selector when provisioning a new PV for this StorageClass.
```

Comment:
I like structured form, but I see some difficulties, too.

Who writes these into StorageClass?  That is an admin object.  If the admin has to write these, the list could potentially get long and tedious.  Is that a problem?  If the provisioner is going to write this into StorageClass, how do we let the admin have final say over what is and is-not allowed?  It seems reasonable that an admin might want to disallow some values or even some whole keys.

displayName and description are awkward.  Storing them in the API makes internationalization difficult.  Any non-english client will have to i18n them, which means nobody should rely on the API strings in software, or those strings become the i18n keys, which means they can never ever change.  Do we have other precedents of this in our API?

What about cases where the default value is "chosen by the implementation" e.g. one of the zones available to the cluster.